### ISSUE TITLE
Why websocket closes after few hours ?

### ISSUE THREAD

**Issue by rawmaterials** [blockchain/api-v1-client-node#44]
i am trying get unconfirmed trasnactions using websocket., but it keeps closing after few hours .

why is that ?

i am using nodejs. 

should i send {"op":"ping"} commands ?
if yes, then how often ?

thanks

**Comment by jtormey** [comment]
Sending ping messages should help, you can do that by calling `socket.op('ping')`. You can also detect when the socket closes and reopen it, for example: `socket.on('close', () => /* reopen socket */)`. Let me know if that helps.

**Comment by rawmaterials** [comment]
i tried sending ping message, every 30 seconds, but still server closed socket after few hours.

i even tried sending unconfirmed_sub , every 30 seconds., and still server closed after few hours.

ill try this method.

socket.on('close', () => /* reopen socket */).

**Comment by mayoralito** [comment]
@rawmaterials did you resolve your problem? If you don't mind, please share with the community your findings/solutions about this matter. Thank you!

@jtormey thanks for your inputs, really appreciate it!

**Comment by rawmaterials** [comment]
yes, i did resolve that problem. as i didnt have control over the websocket, and websocket was closing data after some random hours.,

so i used to quit the websocket connection every hour and then did reconnect.

that worked great for me.
